cluster-autoscaler/simulator/clustersnapshot/basic.go

```diff
@@ -258,36 +254,33 @@ func (snapshot *BasicClusterSnapshot) IsPVCUsedByPods(key string) bool {
 }
 
 // Fork creates a fork of snapshot state. All modifications can later be reverted to moment of forking via Revert()
-// Forking already forked snapshot is not allowed and will result with an error.
-func (snapshot *BasicClusterSnapshot) Fork() error {
-	if snapshot.forkedData != nil {
-		return fmt.Errorf("snapshot already forked")
-	}
-	snapshot.forkedData = snapshot.baseData.clone()
-	return nil
+func (snapshot *BasicClusterSnapshot) Fork() {
+	forkData := snapshot.getInternalData().clone()
+	snapshot.data = append(snapshot.data, forkData)
 }
 
 // Revert reverts snapshot state to moment of forking.
-func (snapshot *BasicClusterSnapshot) Revert() error {
-	snapshot.forkedData = nil
-	return nil
+func (snapshot *BasicClusterSnapshot) Revert() {
+	if len(snapshot.data) == 1 {
+		return
+	}
+	snapshot.data = snapshot.data[:len(snapshot.data)-1]
 }
 
 // Commit commits changes done after forking.
 func (snapshot *BasicClusterSnapshot) Commit() error {
-	if snapshot.forkedData == nil {
+	if len(snapshot.data) <= 1 {
 		// do nothing
 		return nil
 	}
-	snapshot.baseData = snapshot.forkedData
-	snapshot.forkedData = nil
+	snapshot.data = append(snapshot.data[:len(snapshot.data)-2], snapshot.data[len(snapshot.data)-1])
```

Comment:
Why not just `snapshot.data = snapshot.data[:len(snapshot.data)-1]` like you do in revert? Is there some subtle difference in behavior between the two?

Reply:
My understanding of commit() is following: if we have a chain of snapshots A -> B -> C -> D (D is the last fork), then after commit we would like to have A -> B -> D. So we want to have everything up to len-2 (excluding) + the last element. 
Reverting is definitely what we want to have as result for commit()

Reply:
Oh, right, we're deleting last-but-one element. Makes perfect sense, sorry, my comment makes no sense :(